math/src/elliptic_curve/short_weierstrass/curves/bls12_381/pairing.rs

```diff
@@ -152,35 +262,80 @@ fn add_accumulate_line(
         ]),
     ]);
 }
-/// Implements the miller loop for the ate pairing of the BLS12 381 curve.
-/// Based on algorithm 9.2, page 212 of the book
-/// "Topics in computational number theory" by W. Bons and K. Lenstra
+
+// To understand more about how to reduce the final exponentiation
+// read "Efficient Final Exponentiation via Cyclotomic Structure for
+// Pairings over Families of Elliptic Curves" (https://eprint.iacr.org/2020/875.pdf)
+
 #[allow(unused)]
-fn miller(
-    q: &ShortWeierstrassProjectivePoint<BLS12381TwistCurve>,
-    p: &ShortWeierstrassProjectivePoint<BLS12381Curve>,
+pub fn final_exponentiation(
```

Comment:
Similarly, if both implementations are essentially the same in intent, reserve the plain name for the fast version.